include/internal/libfreenect2/packet_processor.h

```diff
@@ -48,11 +138,32 @@ class PacketProcessor
    */
   virtual bool ready() { return true; }
 
+  virtual bool good() { return true; }
+
   /**
    * A new packet has arrived, process it.
    * @param packet Packet to process.
    */
   virtual void process(const PacketT &packet) = 0;
+
+  virtual Allocator *getAllocator() { return &default_allocator_; }
```

Comment:
I see the point in adding this method, but shouldn't it be protected? users of packetprocessor only call allocateBuffer and releaseBuffer, right?